### ISSUE TITLE
Support for macOS Cocoa app

### ISSUE THREAD

**Issue by betzerra** [vapor/vapor#1940]
**TLDR; I'm curious if someone tried to run a macOS Cocoa app with Vapor on it.**

Hi everyone, 

This is a similar question to https://github.com/vapor/vapor/issues/108 but centered in **macOS** instead of **iOS**.

I've been able to create a Framework with my backend logic with Swift Package Framework and then importing it on my existing macOS app by adding it as a sub-project.

I had a couple of link errors at first but after a while I was able to run it. The bad news is that I have [this error from vapor/console](https://github.com/vapor/console/blob/d125366bd9b0a3530f93ab0e27e570e2b9c98dcf/Sources/ConsoleKit/Command/Run/CommandContext.swift#L88-L93)

> "⚠️ [CommandError.excessInput: Too many arguments or unsupported options were supplied: [\"-NSDocumetRevisiosDebugMode\", \"YES\"]]"

I'm not specifically asking about how to resolve this issue, I'm actually more interested in knowing if making this more _"Cocoa friendly"_ is in the roadmap or if there's some blog post about making this work.

**Comment by calebkleveter** [comment]
I know you said you didn't need to issue fixed, but I have a fix anyway 😄. If you updated your Console version to 3.1.1 to apply [PR 93](https://github.com/vapor/console/pull/93), you should be sorted out.

Back to your actual question:

I have personally never seen anyone try to add Vapor to a macOS project. I think a couple people have asked before, but never went anywhere with it when they heard that it isn't officially supported.

We would love to actually support iOS and macOS at some point, and we are getting closer. Vapor 4 will remove a most if not all of the dependencies on APIs that aren't supported on those platforms. But even with that, I don't think much work will be actively done in the area until a third-party (like you 😉) decides to add support or SPM is officially integrated into Xcode.

Something like the [swift-xcode](https://github.com/SwiftXcode/swift-xcode) tool might help.

**Comment by tanner0101** [comment]
Yup @calebkleveter is right, `Environment.detect()` sounds like the problem there. If there are any other issues you run into, we're happy to fix them. :)

**Comment by Sombre-Osmoze** [comment]
Hello @betzerra 
I have vapor running in a macOS app 
Your **error** it's cause by the **Documents Versions debugger** 
<img width="1440" alt="Capture d’écran 2019-04-20 à 05 09 58" src="https://user-images.githubusercontent.com/32064217/56451158-a6d0f880-632a-11e9-98e8-17d0a5f97c87.png">

**You can remove it** in the panel showed in the previous picture 

Or simply use Vapor **like this** 
<img width="1262" alt="Capture d’écran 2019-04-20 à 05 08 56" src="https://user-images.githubusercontent.com/32064217/56451210-edbeee00-632a-11e9-80a7-a5826517836b.png">
